pyccl/halos/halo_model.py

```diff
@@ -96,10 +117,11 @@ def _integ_spline(self, fM, lM):
     def _get_ingredients(self, a, cosmo, get_bf):
         # Compute mass function and bias (if needed) at a new
         # value of the scale factor.
+        _rho0 = cosmo.rho_x(1., "matter", is_comoving=True)
```

Comment:
Can you move this to after the if statement? Otherwise we're always calling this line (and `_get_ingredients` can get called very often!).

Reply:
done